Bottom: a7d9767a8aa68774ee1aa112b3f63d61fea5dd2c
Top:    4e7eadfc2eca9275d876db1d11f37926a8292ed9
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-20 23:59:11 +0100

Refresh of 3d-mouse-cursor-and-other-proj

---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 79ba618..39a65c9 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -1498,7 +1498,10 @@ ghid_draw_everything (BoxTypePtr drawn_area)
   /* FIXME: This isn't strictly correct, as I've ignored the matrix
             elements for homogeneous coordinates. */
   /* NB: last_modelview_matrix is transposed in memory! */
-  reverse_layers = last_modelview_matrix[2][2] < 0;
+  reverse_layers = (last_modelview_matrix[2][2] < 0);
+  if (Settings.ShowSolderSide)
+    reverse_layers = !reverse_layers;
+
   Settings.ShowSolderSide = reverse_layers ? !Settings.ShowSolderSide : Settings.ShowSolderSide;
 
   memset (do_group, 0, sizeof (do_group));
